parsers/MN.py

```diff
@@ -93,23 +97,25 @@ def fetch_production(
     if target_datetime:
         raise NotImplementedError("This parser is not yet able to parse past dates.")
 
-    query_data = query(session)
+    query_data = query(session, logger)
 
-    # Calculated 'unknown' production from available data (consumption, import, solar, wind).
+    # Calculated 'unknown' production from available data (consumption, import, solar, wind, tpp).
     # 'unknown' consists of 92.8% coal, 5.8% oil and 1.4% hydro as per 2020; sources: IEA and IRENA statistics.
-    query_data["unknownMW"] = round(
+    query_data["leftoverMW"] = round(
         query_data["consumptionMW"]
         - query_data["importMW"]
         - query_data["solarMW"]
-        - query_data["windMW"],
+        - query_data["windMW"]
+        - query_data["thermal"],
         13,
     )
 
-    prod_mix = ProductionMix(
-        solar=query_data["solarMW"],
-        wind=query_data["windMW"],
-        unknown=query_data["unknownMW"],
-    )
+    prod_mix = ProductionMix()
+    prod_mix.add_value("solar", query_data["solarMW"])
+    prod_mix.add_value("wind", query_data["windMW"])
+    prod_mix.add_value(
+        "unknown", query_data["leftoverMW"] + query_data["thermal"]
+    )  # Thermal is currently unknown
```

Comment:
There shuld be no math operations for adding the values here, prefarable we should just loop over all the values so it is easy to add new keys/modes in the future.

This does work though so it's not a blocker.

Reply:
I think I've understood what you meant here, but please point out if not